libkbfs/block_journal.go

```diff
@@ -860,65 +860,107 @@ func (j *blockJournal) saveBlocksUntilNextMDFlush() error {
 	return nil
 }
 
-func (j *blockJournal) onMDFlush(ctx context.Context) error {
+// onMDFlush removes at most `maxToRemove` blocks from the
+// `saveUntilMDFlush` journal if one exists.  If `lastToRemove` is
+// zero, it flushes the complete `saveUntilMDFlush` if it has fewer
+// than `maxToRemove` entries; if it doesn't flush the entire journal,
+// it returns the ordinal of the current last entry.  If
+// `lastToRemove` is non-zero, it only flushes up to the minimum of
+// `lastToRemove` and the earliest entry + `maxToRemove`; if this
+// flushes the entire saved journal, it returns 0, otherwise it
+// returns `lastToRemove`.
```

Comment:
Worth mentioning that this is meant to be called repeatedly until 0 is returned.